### PR TITLE
fix `config` option could not overwrite kubelet flags

### LINKED ISSUE #79421: kubelet config file, resolverConfig option not used
**What happened**:

When using kubelet config file, `resolverConfig` option is not used by kubelet.

**What you expected to happen**:

kubelet config option `resolverConfig` is used to set resolver configuration file by kubelet.

**How to reproduce it (as minimally and precisely as possible)**:

kubelet config file:
```
apiVersion: kubelet.config.k8s.io/v1beta1
kind: KubeletConfiguration
[...]
resolverConfig: /var/lib/kubelet-config/resolv.conf
```

During kubelet start, default `/etc/resolv.conf` is used:
```
Jun 26 13:02:53 dev-k8s-worker-r3 kubelet[30137]: I0626 13:02:53.550870   30137 flags.go:33] FLAG: --resolv-conf="/etc/resolv.conf"
```

I confirmed it examining `resolv.conf` inside container, content is taken from node `/etc/resolv.conf` instead of `/var/lib/kubelet-config/resolv.conf`.

On the other hand when using deprecated kubelet parameter `--resolv-conf=/var/lib/kubelet-config/resolv.conf` everything works fine, kubelet is using proper path:

```
Jun 26 13:06:49 dev-k8s-worker-r3 kubelet[6501]: I0626 13:06:49.235306    6501 flags.go:33] FLAG: --resolv-conf="/var/lib/kubelet-config/resolv.conf"
```

**Environment**:
- Kubernetes version (use `kubectl version`):
```
Client Version: version.Info{Major:"1", Minor:"14", GitVersion:"v1.14.3", GitCommit:"5e53fd6bc17c0dec8434817e69b04a25d8ae0ff0", GitTreeState:"clean", BuildDate:"2019-06-06T01:44:30Z", GoVersion:"go1.12.5", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"14", GitVersion:"v1.14.3", GitCommit:"5e53fd6bc17c0dec8434817e69b04a25d8ae0ff0", GitTreeState:"clean", BuildDate:"2019-06-06T01:36:19Z", GoVersion:"go1.12.5", Compiler:"gc", Platform:"linux/amd64"}
```
- Cloud provider or hardware configuration: bare-metal
- OS (e.g: `cat /etc/os-release`): Ubuntu 18.04.1
- Kernel (e.g. `uname -a`):
```
Linux dev-k8s-deployer-r1 4.15.0-42-generic #45-Ubuntu SMP Thu Nov 15 19:32:57 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux
```

### PR DESCRIPTION
**Pull request by zouyee**
**What type of PR is this?**
/kind bug

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #79421

**Special notes for your reviewer**:
When using kubelet config file, `config` option is not used by kubelet.

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: cmd/kubelet/app/server.go [hunk 1]
```diff
@@ -239,7 +237,8 @@ HTTP server: The kubelet can also listen for HTTP and respond to a simple API
 					}
 				}
 			}
-
+			// load kubelet config file which will overwrite kubeletConfig
```
Comment (mattjmcnaughton) [inline cmd/kubelet/app/server.go]: Very possible I'm missing something, but I'm not seeing how calling `PrintFlags` calls us to overwrite `kubeletConfig`. If it is the case that is overwriting `kubeletConfig`, I think this is a poorly named method, and we will want to think of either how the method could not overwrite `kubeletConfig` or how the method name could make it clear that it has destructive side effects.

File: cmd/kubelet/app/server.go [hunk 2]
```diff
@@ -192,7 +190,8 @@ HTTP server: The kubelet can also listen for HTTP and respond to a simple API
 
 			// load kubelet config file, if provided
 			if configFile := kubeletFlags.KubeletConfigFile; len(configFile) > 0 {
-				kubeletConfig, err = loadConfigFile(configFile)
+				config, err := loadConfigFile(configFile)
```
Comment (mattjmcnaughton) [inline cmd/kubelet/app/server.go]: Can you please explain more about how:

```go
config, err := loadConfigFile(configFile)
*kubeletConfig = *config
```
is different than
```go
kubeletConfig, err = loadConfigFile
```

**Review by mattjmcnaughton — COMMENTED** [review]
A couple high level questions from me :) Thanks for taking a look at this issue!

File: cmd/kubelet/app/server.go (hunk 2 above)
Comment (zouyee) [reply cmd/kubelet/app/server.go]: `kubeletConfig, err = loadConfigFile` change the point of `kubeletConfig`

Comment (mattjmcnaughton) [reply cmd/kubelet/app/server.go]: Hmm, I'm afraid I'm still not following. Perhaps @yujuhong or @yifan-gu can weigh in?

**Comment by mtaufen** [comment]
I'm not very certain what the purpose of this PR is.

It looks like the intent may be to log the values from the config file as the flag values in the startup log message, after the config file has been read (in cases where a value is just set via config, I think the flag default will be logged instead of the config file). If that's the case, I think it would be clearer to leave the flag logging where it was (just after parsing the flags), and also log the final config structure once it's loaded.

We _may_ also want to modify the flag logging to indicate whether the flag was set on the command line or not (or even skip logging if not set), since this will help folks determine which flags may be overriding config.

**Pull request closed without merging**